web-app/js/portal/ui/openlayers/layer/NcWMS.js

```diff
@@ -102,9 +102,7 @@ OpenLayers.Layer.NcWMS = OpenLayers.Class(OpenLayers.Layer.WMS, {
         });
     },
 
-    _timeSeriesDatesLoaded: function(datesWithData) {
```

Comment:
About decoupling the first/last day into a different class, I think it should be decoupled in such a way that the interface will decide what to fetch s.t. the interface can fetch all dates, then decide what days to fetch. It will not be a 5 minute job unfortunately...